app/controllers/application_controller.rb

```diff
@@ -1,31 +1,16 @@
 class ApplicationController < ActionController::Base
-  layout 'application'
 
-  before_action :authenticate_user!
-  before_action :complete_signup_profile
-  before_action :check_if_password_expired
+  include AuthenticateMethods
 
-  fine_print_require :general_terms_of_use, :privacy_policy, unless: :disable_fine_print
+  layout 'application'
 
-  def disable_fine_print
-    request.options? ||
-    contracts_not_required ||
-    current_user.is_anonymous?
-  end
+  before_action :authenticate_user!
+  fine_print_skip :general_terms_of_use, :privacy_policy
 
   def check_if_admin
+    # used in blazer.yml to check if user is authorized for Blazer
     return true if !Rails.env.production?
-    is_admin?
-  end
-
-  def check_if_password_expired
-    return true if request.format != :html || request.options?
-
-    identity = current_user.identity
-    return unless identity.try(:password_expired?)
-
-    flash[:alert] = I18n.t(:"controllers.identities.password_expired")
-    redirect_to(password_reset_path)
+    authenticate_admin! unless current_user&.is_administrator?
```

Comment:
Blazer requires the method to be in application_controller then the method is defined in initalizers/blazer.yml 
I couldn't get it working otherwise.

Reply:
Hmmm... I wonder if `blazer.yml` can simply have

```yaml
before_action_method: authenticate_admin!
```

Based on what's here that seems like it should work; good side effect would be that this blazer-specific method could come out of the `ApplicationController`.

Reply:
I guess that doesn't include the non production check, in which case we could maybe be more explicit with the naming to make a new method here

```ruby
def authenticate_admin_on_production_only!
  return true if !Rails.env.production?
  authenticate_admin!
end
```

`authenticate_admin!` already checks `current_user.is_administrator` so don't need to do that again.

Then `blazer.yml` is

```yaml
before_action_method: authenticate_admin_on_production_only!
```

Reply:
I updated the blazer.yml to use the authenticate_admin! and it works great. Doesn't handle the if production part but that's okay.